modules/aws/src/test/java/org/apache/ignite/spi/encryption/AmazonEncryptionSpiSelfTest.java

```diff
@@ -0,0 +1,173 @@
+
+            encSpi.setMasterKeyName("Unknown key name");
+
+            return null;
+        }, NoSuchMasterKeyException.class);
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testEncryptDecrypt() throws Exception {
+        EncryptionSpi encSpi = spi();
+
+        Serializable k = encSpi.create();
+
+        assertNotNull(k);
+
+        byte[] plainText = "Just a test string to encrypt!".getBytes(UTF_8);
+        byte[] cipherText = new byte[spi().encryptedSize(plainText.length)];
+
+        encSpi.encrypt(ByteBuffer.wrap(plainText), k, ByteBuffer.wrap(cipherText));
+
+        assertNotNull(cipherText);
+        assertEquals(encSpi.encryptedSize(plainText.length), cipherText.length);
+
+        byte[] decryptedText = encSpi.decrypt(cipherText, k);
+
+        assertNotNull(decryptedText);
+        assertEquals(plainText.length, decryptedText.length);
+
+        assertEquals(new String(plainText, UTF_8), new String(decryptedText, UTF_8));
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testMasterKeysDigest() throws Exception {
+        EncryptionSpi encSpi = spi();
+
+        byte[] digest = encSpi.masterKeyDigest();
+
+        encSpi.setMasterKeyName(MASTER_KEY_NAME_2);
+
+        byte[] digest2 = encSpi.masterKeyDigest();
+
+        assertNotNull(digest);
+        assertFalse(Arrays.equals(digest, digest2));
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testKeyEncryptDecrypt() throws Exception {
+        EncryptionSpi encSpi = spi();
+
+        Serializable k = encSpi.create();
+
+        assertNotNull(k);
+
+        checkKeyEncryptDecrypt(encSpi, k);
+
+        encSpi.setMasterKeyName(MASTER_KEY_NAME_2);
+
+        checkKeyEncryptDecrypt(encSpi, k);
+    }
+
+    /** */
+    private void checkKeyEncryptDecrypt(EncryptionSpi encSpi, Serializable k) {
+        byte[] encGrpKey = encSpi.encryptKey(k);
+
+        assertNotNull(encGrpKey);
+        assertTrue(encGrpKey.length > 0);
+
+        Serializable k2 = encSpi.decryptKey(encGrpKey);
+
+        assertEquals(k, k2);
+    }
+
+    /** @return Instance of {@link AmazonEncryptionSpi} with {@link JceMasterKey} master key provider. */
+    @NotNull private AmazonEncryptionSpi<JceMasterKey> spi() throws Exception {
+        AmazonEncryptionSpi<JceMasterKey> encSpi = new AmazonEncryptionSpi<>();
+
+        JceMasterKey key1 = getInstance(retrieveEncryptionKey(),
+            "Test provider.", DEFAULT_MASTER_KEY_NAME, "AES/GCM/NoPadding");
+
+        JceMasterKey key2 = getInstance(retrieveEncryptionKey(),
+            "Test provider.", MASTER_KEY_NAME_2, "AES/GCM/NoPadding");
+
+        MasterKeyProvider<JceMasterKey> provider = (MasterKeyProvider<JceMasterKey>)buildMultiProvider(key1, key2);
+
+        encSpi.setProvider(provider);
+
+        encSpi.setMasterKeyName(DEFAULT_MASTER_KEY_NAME);
+
+        GridTestUtils.invoke(encSpi, "onBeforeStart");
```

Comment:
`onBeforeStart` is public. We can call it directly.

Reply:
Done